project-dashboard/scripts/db.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * An abstraction layer for IndexedDB
+ */
+
+const name = 'OppiaProjectDashboard';
+const version = 1; // long long; incrementing fires onupgradeneeded event
+let db = null;
+
+/**
+ * Open IndexedDB
+ * @returns {boolean}
+ */
+const open = async () => {
+  let handle = new Promise((resolve, reject) => {
+    let req = indexedDB.open(name, version);
+    req.onsuccess = event => resolve(event.target.result);
+    req.onupgradeneeded = (event) => {
+      let db = event.target.result;
+
+      // Create keystore
+      if (!db.objectStoreNames.contains('keystore')) {
+        let keystore = db.createObjectStore('keystore', { autoIncrement: true, keyPath: 'name' });
+        keystore.createIndex('name', 'name', { unique: true });
+      }
+    
+      // Create datastore
+      if (!db.objectStoreNames.contains('datastore')) {
+        let datastore = db.createObjectStore('datastore', { autoIncrement: true });
+        // TODO: Create indexes
+      }
+
+      resolve(db);
+    };
+    req.onerror = reject;
+    req.onblocked = reject;
+  });
+
+  db = await handle;
+  return true;
+};
+
+/**
+ * Close IndexedDB
```

Comment:
Here & elsewhere documentation nits (for parity with Oppia web frontend):
- Punctuation at the end of the summary fragments & the param/return lines (when there's a description).
- Empty doc line between the summary & list of documentation tags.